# HRUH Bioinformatics

[How to Atlas]()

* [Atlas basics](atlas.md)
* [Basic Linux commands](Basic_linux_commands.md)
* [Submitting jobs](submit_job.md)
* [Typical file storages](file_storages.md)
* [Installing R-packages](rpackages.md)
* [Run R in atlas](run_r_in_atlas.md)
* [Installing Python-packages with conda](conda.md)
* [Directory map and useful links](directory_map.md)

[How to analyze scRNAseq]()

* [scRNAseq basics](scRNAseq.md)
* [scVI basics](scvi.md)
* [scVI Atlas](scvi_atlas.md)
* [Genotype demultiplexing](genotype_demultiplex.md)
* [HTO demultiplexing](hto_demultiplexing.md)
* [Mitochondrial variants](genotype_demultiplex.md)
* [Velocyto](velocyto.md)
* [scvelo](run_scvelo_cropseq_2023.html)
* [CROP-seq](cropseq.md)
* [Detect variants on scRNA-seq data](detect_variants.md)

[How to analyze TCR-seq]()

* [TCRb-seq basics](tcrseq_basics.md)
* [ScTCRab-seq basics](scTCRabseq.md)
* [GLIPH2](gliph.md)
* [TCRGP](tcrgp.md)

[Miscellaneous]()

* [Color scales](color_scales.md)
* [MAGeCK (CRISPR screen analysis)](mageck.md)
* [CSC bucket download](csc_bucket.md)
* [How to TCGA](TCGA_data.md)
* [How to git](git.md)
* [how to submit scRNAseq data to publish](publish_seq_data.md)
* [How to submit other data to publish](submit_data.md)
* [How to install packages](HRUH_clusters.md)

[How to contribute](contribute.md)

[gimmick:theme](bootstrap)
[gimmick:ForkMeOnGitHub](https://github.com/janihuuh/hruh_webpage)
